packages/back-end/src/models/ExperimentSnapshotModel.ts

```diff
@@ -103,7 +110,107 @@ const ExperimentSnapshotModel = mongoose.model<ExperimentSnapshotDocument>(
 
 const toInterface = (
   doc: ExperimentSnapshotDocument
-): ExperimentSnapshotInterface => omit(doc.toJSON(), ["__v", "_id"]);
+): ExperimentSnapshotInterface =>
+  migrateSnapshot(omit(doc.toJSON(), ["__v", "_id"]));
+
+export function migrateSnapshot(
+  orig: LegacyExperimentSnapshotInterface
+): ExperimentSnapshotInterface {
+  const {
+    activationMetric,
+    statsEngine,
+    // eslint-disable-next-line
+    hasRawQueries,
+    // eslint-disable-next-line
+    hasCorrectedStats,
+    // eslint-disable-next-line
+    query,
+    results,
+    regressionAdjustmentEnabled,
+    metricRegressionAdjustmentStatuses,
+    sequentialTestingEnabled,
+    sequentialTestingTuningParameter,
+    queryFilter,
+    segment,
+    skipPartialData,
+    manual,
+    ...snapshot
+  } = orig;
+
+  // Convert old results to new array of analyses
+  if (!snapshot.analyses) {
+    if (results) {
+      const regressionAdjusted =
+        regressionAdjustmentEnabled &&
+        metricRegressionAdjustmentStatuses?.some(
+          (s) => s.regressionAdjustmentEnabled
+        )
+          ? true
+          : false;
+
+      snapshot.analyses = [
+        {
+          dateCreated: snapshot.dateCreated,
+          status: snapshot.error ? "error" : "success",
+          error: snapshot.error,
+          settings: {
+            statsEngine: statsEngine || "bayesian",
+            dimensions: snapshot.dimension ? [snapshot.dimension] : [],
+            pValueCorrection: null,
+            regressionAdjusted,
+            sequentialTesting: !!sequentialTestingEnabled,
+            sequentialTestingTuningParameter:
+              sequentialTestingTuningParameter ||
+              DEFAULT_SEQUENTIAL_TESTING_TUNING_PARAMETER,
+          },
+          results,
+        },
+      ];
+    } else {
+      snapshot.analyses = [];
+    }
+  }
+
+  // Figure out status from old fields
+  if (!snapshot.status) {
+    snapshot.status = snapshot.error
+      ? "error"
+      : snapshot.analyses.length > 0
+      ? "success"
+      : "running";
+  }
+
+  // Migrate settings
+  // We weren't tracking all of these before, so just pick good defaults
+  if (!snapshot.settings) {
+    snapshot.settings = {
+      manual: !!manual,
+      dimensions: snapshot.dimension
+        ? [
+            {
+              id: snapshot.dimension,
+            },
+          ]
+        : [],
+      // TODO: figure out metrics from results?
+      goalMetrics: [],
+      guardrailMetrics: [],
+      activationMetric: activationMetric ? { id: activationMetric } : null,
+      regressionAdjustmentEnabled: true,
+      startDate: snapshot.dateCreated,
```

Comment:
Are these fields just placeholders? We could use `snapshot.phase` and go look up the values to make a better guess, but maybe for the JIT we don't care about doing that

Same for `experimentId` below. Can't we use `snapshot.experiment`?

Reply:
I don't think these fields affect the behavior of the snapshot at all, it's more just meta info.  I'd like to keep the migration fast and avoid any db lookups or dependencies if possible.  All newly created snapshots should have these fields set already, so doesn't seem worth it in the migration.